packages/engine/Source/Scene/Fog.js

```diff
@@ -145,18 +122,13 @@ Fog.prototype.update = function (frameState) {
   }
 
   const height = positionCartographic.height;
-  const i = findInterval(height);
-  const t = CesiumMath.clamp(
-    (height - heightsTable[i]) / (heightsTable[i + 1] - heightsTable[i]),
-    0.0,
-    1.0,
-  );
-  let density = CesiumMath.lerp(densityTable[i], densityTable[i + 1], t);
-
-  // Again, scale value to be in the range of densityTable (prevents divide by zero) and change to new range.
-  const startDensity = this.density * 1.0e6;
-  const endDensity = (startDensity / tableStartDensity) * tableEndDensity;
-  density = density * (startDensity - endDensity) * 1.0e-6;
+  let density =
+    this.density *
+    this.fogScalar *
+    Math.pow(
+      Math.max(height / this.maxHeight, 1e-6),
```

Comment:
```suggestion
      Math.max(height / this.maxHeight, CesiumMath.EPSILON6),
```

Reply:
What happens when `this.maxHeight` is `0`?

Reply:
> What happens when `this.maxHeight` is 0?

It functionally turns off fog unless you're underground because of the check at the beginning of this function for if you're in space. If you're under height 0 the max value kicks in here and the divide by 0 doesn't seem to be a problem

Reply:
Conceptually I agree! In practice though, I think there is a case where, if `maxHeight` is `0` and `height` happens to be `0`, we can get a value of `NaN`. Can we prevent that?